Form/Type/BasePickerType.php

```diff
@@ -49,11 +51,18 @@ public function __construct(MomentFormatConverter $formatConverter, TranslatorIn
         $this->formatConverter = $formatConverter;
         $this->translator = $translator;
 
-        if (null !== $this->translator) {
-            $this->locale = $this->translator->getLocale();
-        } else {
-            @trigger_error('Initializing '.__CLASS__.' without TranslatorInterface is deprecated since 3.0 and will be removed in 4.0.', E_USER_DEPRECATED);
+        /*
+         * NEXT_MAJOR: remove this check
+         */
+        if (null === $this->translator) {
+            @trigger_error(
+                'Initializing '.__CLASS__.' without TranslatorInterface
+                is deprecated since 2.4 and will be remove in 3.0.',
+                E_USER_DEPRECATED
+            );
             $this->locale = \Locale::getDefault();
+        } else {
```

Comment:
You can use an early return statement to avoid this `else` and decrease complexity.